pkg/raft/tracker/progress.go

```diff
@@ -295,7 +315,11 @@ func (pr *Progress) IsPaused() bool {
 // to guarantee that eventually the flow is either accepted or rejected.
 //
 // In StateSnapshot, we do not send append messages.
-func (pr *Progress) ShouldSendMsgApp(last, commit uint64) bool {
+//
+// If advanceCommit is true, it means that MsgApp owns the responsibility of
+// making sure to advance the followers' commit index. If it's false, it means
+// that the responsibility is primarily on MsgHeartbeat.
```

Comment:
nit: the responsibility is still "primarily" on MsgApp, the heartbeats only close the gap in reliability in cases when MsgApp are dropped